### PR TITLE
CF-618 - Feeds UAE: configure HERP/DERP

### PR DESCRIPTION
**Pull request by usnavi**
- enable herp publishing through log4j2.xml
- Update filters to allow delegating mode

### CODE REVIEW

File: templates/client-auth-n.cfg.xml.erb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
 <client-auth xmlns="http://<%= scope.lookupvar('::repose::cfg_namespace_host') %>/repose/client-auth/v1.0">
+<%- if @delegating_enabled -%>
```
Comment (mwhahaha) [inline templates/client-auth-n.cfg.xml.erb]: I think you're targeting the wrong file for this? This already exists in this file see lines 8-12

File: manifests/filter/rackspace_identity_basic_auth.pp [hunk 2]
```diff
@@ -41,6 +46,7 @@
   $filename             = 'rackspace-identity-basic-auth.cfg.xml',
   $identity_service_url = 'https://identity.api.rackspacecloud.com/v2.0/tokens',
   $token_cache_timeout  = '600000',
+  $delegating_enabled   = false,
```
Comment (mwhahaha) [inline manifests/filter/rackspace_identity_basic_auth.pp]: to be consistent with the other auth filter, this should be $delegating

File: manifests/filter/client_auth_n.pp [hunk 3]
```diff
@@ -102,6 +107,7 @@
   $connection_pool_id  = undef,
   $send_all_tenant_ids = undef,
   $token_expire_feed   = undef,
+  $delegating_enabled  = false,
```
Comment (mwhahaha) [inline manifests/filter/client_auth_n.pp]: This is already in this class, see $delegating

File: templates/client-auth-n.cfg.xml.erb (hunk 1 above)
Comment (usnavi) [inline templates/client-auth-n.cfg.xml.erb]: Yeah, the existing lines are in the wrong place.  I'll remove my param and update this file.

File: manifests/filter/container.pp [hunk 4]
```diff
@@ -144,6 +144,10 @@
 # String. String used in the Via header.
 # Defaults to <tt>undef</tt>
 #
+# [*herp_enabled*]
```
Comment (gregswift) [inline manifests/filter/container.pp]: to be consistent with $delegating this should probably just be $herp

File: templates/client-auth-n.cfg.xml.erb (hunk 1 above)
Comment (gregswift) [inline templates/client-auth-n.cfg.xml.erb]: you also didn't asign quality like alex did. so bear that in mind with whatever you change

File: templates/highly-efficient-record-processor.cfg.xml.erb [hunk 5]
```diff
@@ -7,7 +7,6 @@
     <template crush="<%= @template_crush %>">
 <%- if @template.nil? -%>
     <![CDATA[
-        <?xml version="1.0"?>
```
Comment (gregswift) [inline templates/highly-efficient-record-processor.cfg.xml.erb]: ?

Comment (usnavi) [inline templates/highly-efficient-record-processor.cfg.xml.erb]: Cloud Feeds doesn't allow this when flume attempts to post to it.

**Comment by usnavi** [comment]
@mwhahaha @gregswift - updated as per @mwhahaha comments.

**Comment by usnavi** [comment]
updated as per @gregswift comments.

File: templates/highly-efficient-record-processor.cfg.xml.erb (hunk 5 above)
Comment (mwhahaha) [inline templates/highly-efficient-record-processor.cfg.xml.erb]: Yea this goes back to how this block is actually used by the herp filter, it's actually wrapped in an ATOM payload elsewhere so this is just a block of xml

File: templates/rackspace-identity-basic-auth.cfg.xml.erb [hunk 6]
```diff
@@ -4,3 +4,7 @@
     xmlns="http://<%= scope.lookupvar('::repose::cfg_namespace_host') %>/repose/rackspace-identity-basic-auth/v1.0"
     rackspace-identity-service-uri="<%= @identity_service_url %>"
     token-cache-timeout-millis="<%= @token_cache_timeout %>"/>
```
Comment (mwhahaha) [inline templates/rackspace-identity-basic-auth.cfg.xml.erb]: fix the />, or you'll have a bad time

File: templates/highly-efficient-record-processor.cfg.xml.erb (hunk 5 above)
Comment (gregswift) [inline templates/highly-efficient-record-processor.cfg.xml.erb]: okay.. i just didnt see any context for why it was coming out.

**Comment by usnavi** [comment]
@gregswift - updated as per @mwhahaha XML comment.

**Comment by rackergs** [comment]
Accidentally canceled the wrong build request... closing & reopening to force a refresh

**Comment by gregswift** [comment]
@usnavi tests are failing

**Comment by usnavi** [comment]
@gregswift  - updated the PR - once it turns green i'll ping you again.

**Comment by usnavi** [comment]
@gregswift  - PR is green!

**Pull request merged**